sphinx/domains/cpp.py

```diff
@@ -5495,11 +5514,14 @@ def _parse_parameters_and_qualifiers(self, paramMode: str) -> ASTParametersQuali
         initializer = None
         self.skip_ws()
         if self.skip_string('noexcept'):
-            exceptionSpec = 'noexcept'
-            self.skip_ws()
-            if self.skip_string('('):
-                self.fail('Parameterised "noexcept" not yet implemented.')
-
+            if self.skip_string_and_ws('('):
+                expr = self._parse_expression()
```

Comment:
Please change this to use ``_parse_constant_expression()``.

Reply:
Was modeled according to `def _parse_unary_expression` (line 5031). Should it be changed there as well?